File: Frontend/src/types/ResponseModel/ApplicationTypeResponse.ts
export type ApplicationTypeResp = {
  id: string
  englishName?: string
  vietnameseName?: string
}
export type ApplicationTypeResponse = {
  id: string
  englishName: string
  vietnameseName: string
  totalApplications: number
  createdAt: Date
  updatedAt: Date
}

export type ApplicationTypeListResponse = {
  totalItems: number
  pageSize: number
  totalPages: number
  pageIndex: number
  hasPreviousPage: boolean
  hasNextPage: boolean
  items: ApplicationTypeResponse[]
}

File: Backend/StudentCareSystem.Application/Commons/Models/ApplicationTypes/ApplicationTypeResponse.cs
namespace StudentCareSystem.Application.Commons.Models.ApplicationTypes;

public class ApplicationTypeResponse
{
    public Guid Id { get; set; }
    public string EnglishName { get; set; } = string.Empty;
    public string VietnameseName { get; set; } = string.Empty;
    public int TotalApplications { get; set; }
    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}
